src/JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -124,7 +124,8 @@ public function normalize(mixed $object, ?string $format = null, array $context
             unset($context['operation'], $context['operation_name']);
         }
 
-        if (true === ($context['output']['gen_id'] ?? true) && true === ($context['force_iri_generation'] ?? true) && $iri = $this->iriConverter->getIriFromResource($object, UrlGeneratorInterface::ABS_PATH, $context['operation'] ?? null, $context)) {
+        $operation = $context['operation'] ?? null;
+        if (true === ($context['output']['gen_id'] ?? true) && true === ($context['force_iri_generation'] ?? true) && $iri = $this->iriConverter->getIriFromResource($object, $operation?->getUrlGenerationStrategy() ?? UrlGeneratorInterface::ABS_PATH, $operation, $context)) {
```

Comment:
this is probably not a necessary patch but it's probably better then the previous code